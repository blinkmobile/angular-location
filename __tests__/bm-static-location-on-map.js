/* @flow */
'use strict'

// see: https://docs.angularjs.org/guide/unit-testing
// see: http://facebook.github.io/jest/docs/tutorial-jquery.html

const ng = require('angular')
require('angular-mocks')

require('../lib/index.js')

beforeEach(ng.mock.module('bmLocation'))

let $compile, $rootScope
beforeEach(ng.mock.inject((_$compile_, _$rootScope_) => {
  $compile = _$compile_
  $rootScope = _$rootScope_
}))

test('no attributes', () => {
  const html = '<bm-static-location-on-map></bm-static-location-on-map>'

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('$rootScope.googleMapsApiKey, [coords], [height], [width], [zoom]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const coords = JSON.stringify({ latitude: -35, longitude: 150 })

  const html = `
  <bm-static-location-on-map
    coords='${coords}'
    height="200"
    width="200"
    zoom="10"
  ></bm-static-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('invalid [coords]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const html = `
  <bm-static-location-on-map
    coords="abc"
  ></bm-static-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})
