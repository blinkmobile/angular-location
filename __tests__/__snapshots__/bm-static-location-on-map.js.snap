exports[`test $rootScope.googleMapsApiKey, [coords], [height], [width], [zoom] 1`] = `
"
  <figure>
    <img ng-src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;scale=2&amp;size=200x200&amp;zoom=10&amp;markers=color%3Ared%7C-35%2C150\" alt=\"map with center at -35 latitude, 150 longitude\" height=\"200\" width=\"200\" src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;scale=2&amp;size=200x200&amp;zoom=10&amp;markers=color%3Ared%7C-35%2C150\">
  </figure>
"
`;

exports[`test invalid [coords] 1`] = `
"
  <figure>
    <img ng-src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=0%2C0&amp;scale=2&amp;size=300x300&amp;zoom=0\" alt=\"map of unspecified location\" height=\"300\" width=\"300\" src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=0%2C0&amp;scale=2&amp;size=300x300&amp;zoom=0\">
  </figure>
"
`;

exports[`test no attributes 1`] = `
"
  <figure>
    <img ng-src=\"https://maps.googleapis.com/maps/api/staticmap?key=undefined&amp;center=0%2C0&amp;scale=2&amp;size=300x300&amp;zoom=0\" alt=\"map of unspecified location\" height=\"300\" width=\"300\" src=\"https://maps.googleapis.com/maps/api/staticmap?key=undefined&amp;center=0%2C0&amp;scale=2&amp;size=300x300&amp;zoom=0\">
  </figure>
"
`;
